kuma/authkeys/models.py

```diff
@@ -47,7 +46,9 @@ def generate_secret(self):
         return secret
 
     def check_secret(self, secret):
-        return hash_secret(secret) == self.hashed_secret
+        if not self.hashed_secret:
+            return False
+        return constant_time_compare(hash_secret(secret), self.hashed_secret)
```

Comment:
Yeah, that was a surprise when I saw that, tbh. Since it's the first thing you'd look for when building a token based system.